### ISSUE TITLE
Support stroke cap for grid and tick and domain lines

### ISSUE THREAD

**Issue by donghaoren** [vega/vega-lite#6353]
We should support the strokeCap property for axis grid, tick, and domain lines.

Proposed names:

```ts
axis: {
  gridCap: "butt" | "round" | "square",
  tickCap: "butt" | "round" | "square",
  domainCap: "butt" | "round" | "square"
}
```

Vega axis grid, tick, and domain are all rule marks. We can map these new properties using the "encode" block for Vega axes, and set the rule marks' `strokeCap` property.

**Comment by kanitw** [comment]
I'll fix this in Vega instead -- https://github.com/vega/vega/pull/2534.

**Comment by kanitw** [comment]
This is already in VG 5.11 branch and will be fixed in #6391.
